dcpy/builds/plan.py

```diff
@@ -96,19 +88,24 @@ def plan_recipe(recipe_path: Path, version: str | None = None) -> Recipe:
     recipe: Recipe = recipe_from_yaml(recipe_path)
 
     # Determine the recipe version
-    if version is None:
-        if recipe.version_strategy is None and recipe.version is None:
-            raise Exception("No version provided")
-        elif recipe.version_strategy == VersionStrategy.bump_latest_release:
-            if recipe.version_type is None:
-                raise Exception("Recipe needs a 'version_type' to bump")
-            prev_version = publishing.get_latest_version(recipe.product)
-            recipe.version = versions.bump(
-                prev_version, bumped_part=recipe.version_type
-            )
-        if recipe.version_strategy == VersionStrategy.first_of_month:
-            recipe.version = versions.first_of_month()
-    else:
+    if version is None and recipe.version is None:
+        match recipe.version_strategy:
+            case None:
+                raise Exception("No version provided")
+            case versions.SimpleVersionStrategy.first_of_month:
+                recipe.version = versions.FirstOfMonth.generate().label
+            case versions.SimpleVersionStrategy.bump_latest_release:
+                recipe.version = versions.bump(
+                    previous_version=publishing.get_latest_version(recipe.product),
+                    bump_type=recipe.version_type,
+                )
+            case versions.BumpLatestRelease() as bump:
```

Comment:
These two different "bump latest release"s are a bit clunky. But basically the desired behavior for me was having both of the following be valid
```yml
version_strategy: bump_latest_release
version_strategy:
  bump_latest_release: 2
```

Since DevDB is labeled with quarters, but we always bump by two quarters.

Maybe this is a slightly silly approach in general. This versioning scheme is maybe more dataset-level than recipe-level, and DevDB really should be "biannual" rather than "every two quarters". We could also specify that it prints out in "quarter" format.

Rationale for current implementation is that it's nice to just parse a version string without additional metadata, from a code perspective. 22Q4 is unambiguously a quarter, and it's nice to not have to parse that to "biannual". However, we're never going to be parsing a version on its own, but rather always for a specific data product. Meaning DevDB 22Q4 is pretty impossible to miss as the second release for DevDB of 2022, and we don't necessarily need to specify `bump_latest_release: 2` to know that the next DevDB release will be 23Q2.

I think I've talked myself into reimplementing this. @damonmcc and @sf-dcp would appreciate some input. Might be a bit clearer if you look down at dcpy.utils.versions as well

Reply:
After thinking about it, sort of thinking about leaving as is as a first pass. Need to write a test for `previous` function in publishing. I think we retool as we start to expand dataset-level metadata

Reply:
@alexrichey

Reply:
yeah, I've also gone back and forth on all of that. Another thought: we could just nix the `version_strategy` in the yml, and put it under the version key. So either of these options would all be valid:
```
version: 2023v1
```

```
version: {bump_latest_release: 2}
```